### PR TITLE
Refactor logging in `pkg/metrics` and moves `log` field  to struct field

### PR DESCRIPTION
**Pull request by afzal442**
Signed-off-by: afzal442 <afzal442@gmail.com>

Closes: #2883 

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [ ] release-2.1
  - [ ] release-2.0

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

File: pkg/metrics/chaos-controller-manager.go [hunk 1]
```diff
@@ -19,6 +19,8 @@ import (
 	"context"
 	"reflect"
 
+	z "github.com/chaos-mesh/chaosmonkey/pkg/log"
```
Comment (WangXiangUSTC) [inline pkg/metrics/chaos-controller-manager.go:22]: there is no `chaosmonkey` in this repo

File: pkg/metrics/chaos-controller-manager.go [hunk 2]
```diff
@@ -27,10 +29,9 @@ import (
 	"github.com/chaos-mesh/chaos-mesh/pkg/status"
 )
 
-var log = ctrl.Log.WithName("chaos-controller-manager-metrics-collector")
-
 // ChaosControllerManagerMetricsCollector implements prometheus.Collector interface
 type ChaosControllerManagerMetricsCollector struct {
+	log                 logr.Logger
```
Comment (STRRL) [inline pkg/metrics/chaos-controller-manager.go:34]: please use logger here :)

File: pkg/metrics/chaos-controller-manager.go [hunk 3]
```diff
@@ -19,6 +19,8 @@ import (
 	"context"
 	"reflect"
 
+	z "github.com/chaos-mesh/chaos-mesh/pkg/log"
```
Comment (STRRL) [inline pkg/metrics/chaos-controller-manager.go:22]: please do not use an alias `z` which does not make sense to developers. IMO, importing it without an alias is OK for me .

File: pkg/metrics/chaos-controller-manager.go [hunk 4]
```diff
@@ -53,7 +54,10 @@ func NewChaosControllerManagerMetricsCollector(manager ctrl.Manager, registerer
 		store = manager.GetCache()
 	}
 
+	logger, _ := z.NewDefaultZapLogger()
+
```
Comment (STRRL) [inline pkg/metrics/chaos-controller-manager.go:58]: please do not create another logger, acquire the logger as a parameter in the constructor.

File: pkg/metrics/chaos-daemon.go [hunk 5]
```diff
@@ -59,7 +61,9 @@ type ChaosDaemonMetricsCollector struct {
 
 // NewChaosDaemonMetricsCollector initializes metrics for each chaos daemon
 func NewChaosDaemonMetricsCollector() *ChaosDaemonMetricsCollector {
+	logger, _ := z.NewDefaultZapLogger()
```
Comment (STRRL) [inline pkg/metrics/chaos-daemon.go:64]: ditto, please do not create another logger, acquire the logger as a parameter in the constructor.

**Review by STRRL — CHANGES_REQUESTED** [review]
Thanks for your contribution! ptal

File: pkg/metrics/chaos-controller-manager.go (hunk 2 above)
Comment (afzal442) [reply pkg/metrics/chaos-controller-manager.go:34]: I'm not sure how I use logger here.

Comment (afzal442) [reply pkg/metrics/chaos-controller-manager.go:34]: @Andrewmatilde Any suggestion here?

Comment (Andrewmatilde) [reply pkg/metrics/chaos-controller-manager.go:34]: `logger logr.Logger` is OK.

Comment (STRRL) [reply pkg/metrics/chaos-controller-manager.go:34]: > do you mean like below?
> 
> ```go
> logger logr.Logger
> ```

YES!

Comment (STRRL) [reply pkg/metrics/chaos-controller-manager.go:34]: > I'm not sure how I use logger here.
> 
> becaz `NewDefaultZapLogger()` returns logger and error both,

 You cloud append the `logger` as another parameter in the constructor of `ChaosControllerManagerMetricsCollector`, and do not care about how the `logger` comes here.

**Review by STRRL — APPROVED** [review]
LGTM!

Thanks for your contribution!

**Review by WangXiangUSTC — APPROVED** [review]
LGTM

**Comment by STRRL** [comment]
> cherry pick to release-2.1 failed

I would manually cherry-pick this PR and #3025

**Comment by STRRL** [comment]
> > cherry pick to release-2.1 failed
> 
> I would manually cherry-pick this PR and #3025

There is no requirement to cherry-pick #3025 into release-2.1, because metrics about chaos-daemon was not cherry-picked into release-2.1

**Pull request merged**